src/pymor/operators/block.py

```diff
@@ -245,3 +257,243 @@ def assemble_lincomb(self, operators, coefficients, solver_options=None, name=No
             for i in range(self.num_source_blocks):
                 blocks[i] = self._blocks[i, i] * c
             return self.__class__(blocks)
+
+
+class SecondOrderOperator(BlockOperator):
```

Comment:
I think that this class name is a bit misleading.
Also: Would it be an alternative to implement `apply_inverse` and friends for a generic 2-by-2 `BlockOperator`?

Reply:
I'm not sure if systems with [Schur complements](https://en.wikipedia.org/wiki/Block_matrix#Block_matrix_inversion) can be solved efficiently in the generic case...

But, I guess the same can be said for generic `BlockOperators`. Maybe there is a way to see which Schur complement is easy to solve with and use that one, otherwise use the basic `apply_inverse`? Maybe having `assemble_concatenation` would help?